remote-continuous-query/src/main/java/org/infinispan/tutorial/simple/remote/query/InfinispanRemoteContinuousQuery.java

```diff
@@ -0,0 +1,152 @@
+package org.infinispan.tutorial.simple.remote.query;
+
+import static org.infinispan.query.remote.client.ProtobufMetadataManagerConstants.PROTOBUF_METADATA_CACHE_NAME;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+import java.util.UUID;
+
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.hotrod.Search;
+import org.infinispan.client.hotrod.configuration.ClientIntelligence;
+import org.infinispan.client.hotrod.configuration.ConfigurationBuilder;
+import org.infinispan.client.hotrod.impl.ConfigurationProperties;
+import org.infinispan.client.hotrod.marshall.ProtoStreamMarshaller;
+import org.infinispan.protostream.SerializationContext;
+import org.infinispan.protostream.annotations.ProtoSchemaBuilder;
+import org.infinispan.query.api.continuous.ContinuousQuery;
+import org.infinispan.query.api.continuous.ContinuousQueryListener;
+import org.infinispan.query.dsl.Query;
+import org.infinispan.query.dsl.QueryFactory;
+
+/**
+ * The Remote Continuous Query simple tutorial.
+ *
+ * @author Katia Aresti, karesti@redhat.com
+ */
+public class InfinispanRemoteContinuousQuery {
+
+   // USERS will be used to create random posts
+   private static final List<String> USERS = Arrays.asList(
+         "gustavoalle",
+         "remerson",
+         "anistor",
+         "karesti",
+         "ttarrant",
+         "belen_esteban",
+         "dberindei",
+         "galderz",
+         "wburns",
+         "pruivo",
+         "oliveira",
+         "vrigamonti");
+
+   // HASHTAGS will be used to create random posts
+   private static final List<String> HASHTAGS = Arrays.asList(
+         "love",
+         "instagood",
+         "photooftheday",
+         "fashion",
+         "beautiful",
+         "happy",
+         "cute",
+         "tbt",
+         "like4like",
+         "followme",
+         "infinispan");
+
+   private static final String CACHE_NAME = "instaposts";
+
+   public static void main(String[] args) throws Exception {
+      // Create a configuration for a locally-running server
+      ConfigurationBuilder builder = new ConfigurationBuilder();
+      builder.addServer().host("127.0.0.1")
+            .port(ConfigurationProperties.DEFAULT_HOTROD_PORT)
+            .clientIntelligence(ClientIntelligence.BASIC) // This line is needed to connect with docker4mac
```

Comment:
Hmmm, if the host is `127.0.0.1` do you need client intelligence basic? I'd remove basic intelligence from this tutorial.

Reply:
ok